src/sentry/sentry_apps/token_exchange/grant_exchanger.py

```diff
@@ -65,10 +67,12 @@ def _validate(self) -> None:
         Validator(install=self.install, client_id=self.client_id, user=self.user).run()
 
         if not self._grant_belongs_to_install() or not self._sentry_app_user_owns_grant():
-            raise APIUnauthorized("Forbidden grant")
+            raise SentryAppIntegratorError(APIUnauthorized("Forbidden grant"))
 
         if not self._grant_is_active():
-            raise APIUnauthorized("Grant has already expired")
+            raise SentryAppIntegratorError(
+                APIUnauthorized("Grant has already expired"), status_code=403
```

Comment:
Are we changing the status code for this, or is this just because we're now wrapping the error? Also is it worth using the same enums you're using in testing for these?

Reply:
`APIUnauthorized` has a built in status code of 403, but `SentryAppIntegratorError`'s default status code is 400 and we use the [status code attached](https://github.com/getsentry/sentry/blob/c09e99da1b97a9ad534d4046e0d53bfc0dca0df8/src/sentry/sentry_apps/api/bases/sentryapps.py#L130) to `SentryAppIntegratorError`. 

er I'm a bit confused wdym by `using the same enums you're using in testing for these` like testing we raise a `SentryAppIntegratorError` ?

Reply:
Ahh okay, so it's a side effect of the composition swallowing the status code then 😅 

>er I'm a bit confused wdym by using the same enums you're using in testing for these like testing we raise a SentryAppIntegratorError ?

Oh I just meant the drf `status` import in the test:
`from rest_framework import serializers, status`

https://github.com/getsentry/sentry/blob/dec62be033f0992723ed3116925950d67236ebc8/src/sentry/sentry_apps/api/endpoints/sentry_app_authorizations.py#L85

Reply:
ohhhhh, makes sense 👍